### ISSUE TITLE
void functions should not have a return

### ISSUE THREAD

**Issue by Fank** [palantir/tslint#1383]
### Feature request
- **TSLint version**: `3.12.1`
- **TypeScript version**: `1.8.10`
- **Running TSLint via**: (please choose one) CLI / grunt-tslint / Atom
#### TypeScript code being linted

``` ts
export class MyTestClass {
    public foobar(): void {
        return this._voidTest();
    }

    private _voidTest(): void {
        // Something
    }
}
```
#### Actual behavior

Everything is fine
#### Expected behavior

The `return` in `MyTestClass.foobar` should show an error, because a `void` function/method does not return something.

**Comment by adidahiya** [comment]
closing as I think #1903 mostly covers this use case; let me know if we should re-open
